lcls_naming_tool/lcls_naming_tool.py

```diff
@@ -0,0 +1,244 @@
+
+    pv_name = [x.strip() for x in user_input.split(':')]
+
+    for element in pv_name:
+        if not re.match('^[a-zA-Z0-9]+', element):
+            print('Invalid')
+            return False
+
+    # Check the length of the PV name is valid
+    if (len(pv_name) < 3) or (len(pv_name) > 5):
+        print('Invalid')
+        return False
+
+    # parse the functional component (FFFFF)
+    fc_taxon = list(pv_name[0].strip())
+    fc_valid = functional_component_is_valid(fc_taxon)
+
+    if fc_valid:
+
+        # check for PV name with 3 elements (FFFFF:CCC:XXXX)
+        if (len(pv_name) == 3):
+            ccc_taxon = pv_name[1].strip()
+            ccc_valid = constituent_component_is_valid(ccc_taxon)
+
+            if ccc_valid:
+                print('Valid')
+                return True
+            else:
+                print('Invalid')
+                return False
+
+        # Check for PV name with 4 elements (FFFFF:GGG:CCC:XXXX or FFFFF:CCC:NN:XXXX).
+        if (len(pv_name) == 4):
+            fg_taxon = pv_name[1].strip()
+            fg_valid = fungible_is_valid(fg_taxon)
+
+            if fg_valid:
+                # validate for FFFFF:GGG:CCC:XXXX
+                ccc_taxon = pv_name[2].strip()
+                ccc_valid = constituent_component_is_valid(ccc_taxon)
+
+                cspv_taxon = pv_name[3].strip()
+                cspv_valid = process_variable_is_valid(cspv_taxon)
+                              
+                if ccc_valid and cspv_valid:
+                    print('Valid')
+                    return True
+                else:
+                    print('Invalid')
+                    return False
+            else:
+                # validate for FFFFF:CCC:NN:XXXX
+                ccc_taxon = pv_name[1].strip()
+                ccc_valid = constituent_component_is_valid(ccc_taxon)
+
+                nn_taxon = pv_name[2].strip()
+                nn_valid = increment_is_valid(nn_taxon)
+
+                if ccc_valid and nn_valid:
+                    print('Valid')
+                    return True
+                else:
+                    print('Invalid')
+                    return False
+
+        # check for PV name with 5 elements (FFFFF:GGG:CCC:NN:XXXX)
+        if (len(pv_name) == 5):
+            fg_taxon = pv_name[1].strip()
+            fg_valid = fungible_is_valid(fg_taxon)
+
+            ccc_taxon = pv_name[2].strip()
+            ccc_valid = constituent_component_is_valid(ccc_taxon)
+
+            nn_taxon = pv_name[3].strip()
+            nn_valid = increment_is_valid(nn_taxon)
+
+            if fg_valid and ccc_valid and nn_valid and fc_valid:
+                print('Valid')
+                return True
+            else:
+                print('Invalid')
+                return False
+
+    else:
+        print('Invalid')
+        return False
+
+
+def main():
+
+    load_taxons()
+
+    # Check if user entered a file name or PV name (a string)
+    if not sys.stdin.isatty():
```

Comment:
This is checking if we're in a real terminal or if we're being redirected.  Aka something like:
`python lcls_naming_tool.py` vs `python lcls_naming_tool.py | cat` 

I think there are more direct ways of seeing if the argument supplied is a file or PV string.  We could just check if the provided string exists as a filepath, then branch the logic off of this.

Or perhaps I'm misunderstanding what you want to do here, because this reads to me: "if we're being redirected, use stdin as an input stream and operate interactively"?

Reply:
It's checking if the user is in a terminal. This can be replaced by argparse, will fix in next PR.